src/App.js

```diff
@@ -11,43 +11,74 @@ export class App extends Component{
   constructor() {
     super()
     this.state = {
+      id: 0,
       isGenerator: null,
       numPlayers: 0,
       word: '',
-      guess: '',
-      // attempts: [],
-      attempts: ['cuts', 'butts', 'coconuts'],
+      // guess: '',
+      guesses: [],
+      // attempts: ['cuts', 'butts', 'coconuts'],
       // hints: [],
-      display: ['d', '_', '_', 'o', '_', 'a', 'u', 'r']
+      isOver: false,
+      display: []
+      // display: ['d', '_', '_', 'o', '_', 'a', 'u', 'r']
     }
   }
+
   designateRole = async (role) => {
-    // const isGenerator = await apiCalls.joinGame(role)
-    // this.setState({isGenerator})
-    console.log('role', role)
+    try {
+      const joinGame = await apiCalls.joinGame(role)
+      console.log(joinGame)
+      // dont need numPlayers, instead need boolean "ready"
+      // the below line will get cleaned so we can just setState({joinGame})
+      this.setState({ isGenerator: joinGame.isGen, numPlayers: joinGame.numPlayers, id: joinGame.id })
+    } catch(error) {
+      console.log('error', error)
+    }
+    // console.log('role', role)
   }
+
   makeWordToGuess = async (createdWord) => {
-    // const word = await apiCalls.createWord(createdWord, 1)
-    // this.setState({word})
-    console.log("createdWord", createdWord)
+    try {
+      const word = await apiCalls.createWord(createdWord, this.state.id)
+      this.setState({ word })
+    } catch(error) {
+      console.log("error2", error)
+    }
   }
+
   makeGuess = async (newGuess) => {
-    // const guess = await apiCalls.makeGuess(newGuess)
-    // this.setState({guess})
-    console.log('newGuess', newGuess)
+    try {
+      const guess = await apiCalls.createWord(newGuess, this.state.id)
+      this.setState({ guess })
+    } catch(error) {
+      console.log("error3", error)
+    }
   }
+
+  resetGame = async () => {
+    try {
+      const reset = await apiCalls.clearGame();
+      this.setState({ reset })
+    } catch (error) {
+      console.log(error)
+    }
+  }
+
   render() {
-    if (this.state.numPlayers > 1 && !this.state.word && History[History.length -1] !=='/word-selector') {
-      History.push('/word-selector');
-    } else if (this.state.numPlayers > 1 && this.state.word && History[History.length -1] !=='/gamepage') {
-      History.push('/gamepage');
-    } else if (History[History.length -1] !=='/') {
-      History.push('/');
+    if (this.state.numPlayers > 1 && !this.state.word && History[History.length -1] !== '/word-selector') {
+      window.setTimeout(() => History.push('/word-selector'), 1);
+    } else if (this.state.numPlayers > 1 && this.state.word && History[History.length -1] !== '/gamepage') {
+      
+      window.setTimeout(() => History.push('/gamepage'), 1);
+    } else if (History[History.length -1] !== '/') {
+      window.setTimeout(() => History.push('/'), 1);
```

Comment:
I would like to document why this is here:

We were getting some `multiple rendering` errors. SetTimeout gives necessary async timing to avoid these conflicts. We should make a low priority issue to perhaps clean up this.... somewhat smelly code. I wish I knew how, it may be an adventure.